### ISSUE TITLE
False positive parse error for inline `as` type assertions in $derived expressions

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#84]
## Description

svelte-check-rs 0.7.0 reports a parse error for valid TypeScript `as` type assertions when used inline within `$derived` expressions.

## Reproduction

```svelte
<script lang="ts">
  let export_data = $state(true)

  let export_config = $derived(
    export_data
      ? {
        formats: [`csv`, `json`] as (`csv` | `json`)[],
        filename: `table-export`,
      }
      : null,
  )
</script>
```

## Error

```
src/lib/table/HeatmapTable.svelte:134:9
Error: Unexpected token
https://svelte.dev/e/js_parse_error (js_parse_error)
```

## Expected behavior

This is valid TypeScript/Svelte 5 code and should parse without errors. The official `svelte-check` does not report this as an error.

## Workaround

Extract the typed array to a separate const:

```svelte
<script lang="ts">
  let export_data = $state(true)

  const default_formats: (`csv` | `json`)[] = [`csv`, `json`]
  let export_config = $derived(
    export_data
      ? {
        formats: default_formats,
        filename: `table-export`,
      }
      : null,
  )
</script>
```

## Environment

- svelte-check-rs version: 0.7.0
- Svelte version: 5.48.2
- OS: macOS / Linux (GitHub Actions)
